public/pages/EditFeatures/components/CategoryField/CategoryField.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiText,
+  EuiLink,
+  EuiIcon,
+  EuiFormRow,
+  EuiPage,
+  EuiPageBody,
+  EuiComboBox,
+  EuiCheckbox,
+  EuiTitle,
+  EuiCallOut,
+  EuiSpacer,
+} from '@elastic/eui';
+import { Field, FieldProps } from 'formik';
+import { get, isEmpty } from 'lodash';
+import React, { useState, useEffect } from 'react';
+import ContentPanel from '../../../../components/ContentPanel/ContentPanel';
+// @ts-ignore
+import { toastNotifications } from 'ui/notify';
+import {
+  isInvalid,
+  getError,
+  validateCategoryField,
+} from '../../../../utils/utils';
+//@ts-ignore
+import chrome from 'ui/chrome';
+
+interface CategoryFieldProps {
+  isHCDetector: boolean;
+  categoryFieldOptions: string[];
+  setIsHCDetector(isHCDetector: boolean): void;
```

Comment:
you may delete this method, I think original `CategoryFieldProps` is already good for integration

Reply:
I added this to update the `isHCDetector` flag in `editFeatures`. This was needed to properly validate the form in the case of the box being checked, but no value being set.

Reply:
Basically, since we don't perform validation immediately after enabling the checkbox now, this became an issue that had to be fixed by updating `isHCDetector` in `editFeatures`, so we can properly validate/invalidate in `handleSaveChanges()`.

Reply:
Got it. I thought you may delete it since I already modified code to support validation only when checkbox is enabled. But I think your new implementation is better. Approved.